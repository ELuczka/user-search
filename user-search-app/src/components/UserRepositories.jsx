import React, { useEffect } from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Box, Grid, Link, Paper, Typography } from "@material-ui/core";

import { useAsyncData } from "../api/useAsyncData";
import { fetchUser } from "../requests/fetchUser";
import { CLIENT_ID, CLIENT_SECRET } from "./constances";
import ErrorMessage from "../common-coponents/ErrorMessage";
import Loading from "../common-coponents/Loading";

const useStyles = makeStyles((theme) => ({
    repoLink: {
      padding: theme.spacing(),
      marginTop: theme.spacing(2),
    },
  }));

const UserRepositories = ({ user }) => {
    const classes = useStyles();

  const userLogin = user?.data?.login;

  const url = `https://api.github.com/users/${userLogin}/repos?sort=stars&per_page=3&client_id=${CLIENT_ID}&client_secret=${CLIENT_SECRET}`;

  const { data, isLoading, error, loadData } = useAsyncData({
    loadOnMount: true,
    fetchData: () => fetchUser(url),
  });

  useEffect(() => {
    loadData(url);
  }, [url]);

  const hasRepos = data?.data?.length > 0;

  return (
    <>
      <Grid container item xs={5} direction="column">
        <Grid item xs={6}>
          <Box fontWeight="fontWeightBold">Top repositories</Box>
        </Grid>

        {hasRepos ? (
          data.data.map((el) => (
            <Grid item xs={8} key={el.id}>
              <Paper className={classes.repoLink}>
                <Link href={el.svn_url}>
                  <Typography noWrap>{el.name}</Typography>
                </Link>
              </Paper>
            </Grid>
          ))
        ) : (
          <Typography>This user has no repository.</Typography>
        )}

        <Grid container item xs={5}>
          {isLoading && <Loading />}
          {error && <ErrorMessage errorMessage="Something went wrong... " />}
        </Grid>
      </Grid>
    </>
  );
};

export default UserRepositories;