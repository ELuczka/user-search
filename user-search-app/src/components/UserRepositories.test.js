import React from 'react';

import renderer from 'react-test-renderer';

import UserRepositories from './UserDetails';

const memoizedUserDetails = {
    name: "Test User",
    bio: "Hello World, I am React Developer!",
    avatarUrl: "https://avatars1.githubusercontent.com/u/129108?v=4",
    login: "Test",
  };

it('UserRepositories to match snapshot', () => {
    const component = renderer
    .create(<UserRepositories user={memoizedUserDetails}/>)
    .toJSON();

    expect(component).toMatchSnapshot();
});