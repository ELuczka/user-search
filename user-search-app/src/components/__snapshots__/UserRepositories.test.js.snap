// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 1`] = `
<div
  className="MuiGrid-root makeStyles-container-2 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-flex-end MuiGrid-grid-xs-5"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
  >
    <div
      className="MuiAvatar-root MuiAvatar-rounded makeStyles-avatar-1"
    >
      <img
        alt="Avatar"
        className="MuiAvatar-img"
        src="https://avatars1.githubusercontent.com/u/129108?v=4"
      />
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
  >
    <div
      className="MuiBox-root MuiBox-root-3"
    >
      Test User
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
    >
      <div
        className="MuiBox-root MuiBox-root-4"
      >
        Hello World, I am React Developer!
      </div>
    </div>
  </div>
</div>
`;
